File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder();
        
        for (int i = 1; i < food.length; i++) {
            int shared = food[i] / 2;
            if (shared > 0) sb.append(String.valueOf(i).repeat(shared));
        }
        
        sb.append(0);
        sb.append(new StringBuilder(sb.substring(0, sb.length() - 1)).reverse());
        
        return sb.toString();
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.kt
class Solution {
    fun solution(food: IntArray): String {
        val left = buildString {
            for (i in 1 until food.size) {
                repeat(food[i] / 2) { append(i) }
            }
        }
        
        return left + "0" + left.reversed()
    }
}